src/server/api/controllers/vessel-report.controller.js

```diff
@@ -0,0 +1,77 @@
+"use strict";
+
+const HttpError = require("../lib/utils/http-error");
+const knex = require("../../config/db");
+
+//Get vessels reports by vessel id
+const getVesselsReportByVesselId = async id => {
+  const vesselReportsByVessel = await knex
+    .from("vessel_reports")
+    .where({ vessel_id: id });
+  if (vesselReportsByVessel.length === 0) {
+    return `no vessel exist with that id ${id} `;
+  }
+  return vesselReportsByVessel;
+};
+
+//create vessel_reports
+const createVesselReport = async ({ body }) => {
+  const {
+    vessel_id,
+    position_waypoints,
+    current_speed,
+    hfo_consumption,
+    lsfo_consumption
+  } = body;
+
+  const vessel = await knex
+    .from("vessels")
+    .select("*")
+    .where({
+      id: vessel_id
+    });
+  if (vessel.length === 0) {
+    throw new HttpError("Bad request", `${vessel_id} vessel not found!`, 404);
+  }
+
+  const vessel_reports = await knex("vessel_reports").insert({
```

Comment:
Naming conventions:

DB: vessel_reports
Table names are always plural as they represent a collection of vessel_reports

Javascript: vesselReports (camelCase)
Try to make the variable name reflect whether you have a single record or a collection. So in the example above it should be vesselReport, but in the "get vessel report by ID" it should be singular and in the "get vessel reports by vessel ID" it should be plural (we are getting 0, 1 or many vessel reports matching one given ID as input).

File names: vessel-reports
I think we should keep file names plural consistently. I considered making routes plural and controllers singular, but I'm not sure that's valuable and currently everything is plural (I think?).

URL/URIs: ressel-reports

Reply:
Please check naming conventions throughout your PR. There are many other instances where it can be improved.

Reply:
Renamed the variable and file name as per requested.